grails-app/init/org/pih/warehouse/BootStrap.groovy

```
@@ -658,53 +658,42 @@ class BootStrap {
-            List<String> changelogVersions = resources
-                    .collect { it.filename }
-                    .findAll { it.matches("\\d{1,3}.\\d{1,3}.x") }
-                    .reverse()
+            LiquibaseUtil.executeChangeLog("views/drop-all-views.xml")
 
-            // Exclude the newest changelog version, this one should be run separately
-            List<String> previousChangelogVersions = !changelogVersions.empty ? changelogVersions.tail() : []
+            def executedChangeLogVersions = LiquibaseUtil.getExecutedChangelogVersions()
+            log.info("executedChangelogVersions: " + executedChangeLogVersions)
 
-            //If nothing has been created yet, let's create all new database objects with the install scripts
-            if (!executedChangelogVersions) {
+            // If nothing has been created yet, let's create all new database objects with the install scripts
+            if (!executedChangeLogVersions) {
                 log.info("Running install changelog ...")
-                liquibase = new Liquibase("install/changelog.xml", new ClassLoaderResourceAccessor(), database)
-                liquibase.update(null as Contexts, new LabelExpression());
+                LiquibaseUtil.executeChangeLog("install/changelog.xml")
             }
 
             // Check if the executed changelog versions include one of the previous versions
             // and if so, then we need to keep running the old updates to catch up to 0.9.x
-            boolean hasExecutedAnyPreviousChangesets =
-                executedChangelogVersions.any { previousChangelogVersions.contains(it.version) }
+            Set<String> upgradeChangeLogVersions = LiquibaseUtil.upgradeChangeLogVersions
+            log.info("upgradeChangeLogVersions: " + upgradeChangeLogVersions)
+            if (upgradeChangeLogVersions?.empty) {
+                throw new RuntimeException("Unable to execute database migrations")
+            }
```

Not sure if this is the right approach (or error message) but I'd prefer to throw an error rather than silently failing like we have for the past few months.

---

Let's throw an error, but perhaps it would be better to give more descriptive reason? Especially **why exactly** "unable to execute"

---

The reason I wasn't more specific is because I honestly have no idea how to explain this. :)

Let me know if you have any thoughts.